From 668dde9f2a216c6b2e8e9a4735daf0c3f4c13278 Mon Sep 17 00:00:00 2001
From: Marc Mutz <marc.mutz@qt.io>
Date: Tue, 30 Sep 2025 11:45:24 +0200
Subject: [PATCH 248/553] QWinRegistryKey: don't use qSwap()

We know the type of m_key (it's a pointer), so use qt_ptr_swap()
instead of the expensive-to-compile qSwap() ADL-enabler.

Amends 05a829f923a88e69b2ceaa372820a2dcb8c083cd.

Pick-to: 6.8 6.5
Task-number: QTBUG-97601
Change-Id: I125908270765b6dbb3472f663c410309235a9b72
Reviewed-by: Friedemann Kleint <Friedemann.Kleint@qt.io>
(cherry picked from commit b1a25071a06c5d6dc0e009ea5f0b1f81307290fe)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/corelib/kernel/qwinregistry_p.h | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/corelib/kernel/qwinregistry_p.h b/src/corelib/kernel/qwinregistry_p.h
index 6f23bb4517c..a1a84e8bd24 100644
--- a/src/corelib/kernel/qwinregistry_p.h
+++ b/src/corelib/kernel/qwinregistry_p.h
@@ -41,7 +41,10 @@ public:
     QWinRegistryKey(QWinRegistryKey &&other) noexcept
         : m_key(std::exchange(other.m_key, nullptr)) {}
     QT_MOVE_ASSIGNMENT_OPERATOR_IMPL_VIA_MOVE_AND_SWAP(QWinRegistryKey)
-    void swap(QWinRegistryKey &other) noexcept { qSwap(m_key, other.m_key); }
+    void swap(QWinRegistryKey &other) noexcept
+    {
+        qt_ptr_swap(m_key, other.m_key);
+    }
 
     [[nodiscard]] bool isValid() const { return m_key != nullptr; }
 
-- 
2.51.2

